### PR TITLE
Messages Tests <final>

### LINKED ISSUE #84: Add tests for each messages class
As seen in the other test files
Test each class:
Messages
MessageBoard
MessageTags

add functions to make sure the models are correct.

### PR DESCRIPTION
**Pull request by shakedhagag**
## Tests for each Messages class:

This PR contains tests for each of the messages model class:
1. MessageBoards
2. Messages
3. MessageTags

I have arranged them all and made sure the code pushed is working and passing pytest. 
Note that I had to rebase after I pushed then push again because a User class field was deleted and my files had dependencies with it. (Field: 'Type'). 
I also noticed that pytest is not passing online but locally it does:
image

Only one test doesn't pass online:
``AcadeMeData/test_msgtags.py::TestMessageTagsModel::test_get_msg_tag FAILED``

I wonder why it happens.
 
This PR closes #84

### CODE REVIEW

**Comment by Yarboa** [comment]
sorry i do not understand why you keep adding migration files,
Every thing already done, you are just running test.

Like i mentioned in the other PR remove those files please it is not necessary

**Comment by shakedhagag** [comment]
> sorry i do not understand why you keep adding migration files, Every thing already done, you are just running test.
> 
> Like i mentioned in the other PR remove those files please it is not necessary

Hey please read the PR, there are several changes required in the migration files after rebase. I remembered what you said about migration files and I specificly added these ones and mentioned why in the description.

**Comment by Yarboa** [comment]
> > sorry i do not understand why you keep adding migration files, Every thing already done, you are just running test.
> > Like i mentioned in the other PR remove those files please it is not necessary
> 
> Hey please read the PR, there were several changes required in the migration files after rebase. I remembered what you said about migration files and I specificly added these ones and mentioned why in the description.

Please take a look, you have in migration tests History already
You receive two objects

Think of the test different, in fixture called as parameter you created db object
AcadeMeData/migrations/0007_Degree_test_data.py
In the test itself use the following:

sample code
>>> from AcadeMeData.models import Degree
>>> tt = Degree.get_degree_by_name("History")
>>> tt
<Degree: History>

File: AcadeMeData/test_msgtags.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+@pytest.fixture
+def generate_msgtags(id=1):
+    tag = MessageTags(id, generate_message, user_example)
+    return tag
+
+
+@pytest.mark.django_db
+class TestMessageTagsModel:
+    def test_get_msg_tag(self, generate_msgtags):
+        tag = generate_msgtags
```
Comment (Yarboa) [inline AcadeMeData/test_msgtags.py:31]: You are already calling it in the fixture
Please rethink of the test, what do you want to test here?

File: AcadeMeData/test_msgtags.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        tag_test = MessageTags.get_msg_tag(1)
```
Comment (Yarboa) [inline AcadeMeData/test_msgtags.py:32]: so maybe you should parse the id which is default=1 as a parameter in the test to fixture

Than search that tag id exist, and maybe you could assert one more thing in that tag

Comment (Yarboa) [reply AcadeMeData/test_msgtags.py:32]: def test_get_msg_tag(self, generate_msgtags, id=3):

**Pull request closed without merging**